##  1.Load dataset into a SQL database. 

Loading the Dataset into SQL Database

- Downloaded the Superstore Sales dataset from Kaggle.
- Uploaded the CSV file to Databricks using the Volume storage.
- Created a table named superstore_dataset from the uploaded CSV file.
- Verified that the data was loaded successfully by querying the table and checking the row count.

In [0]:
select * from superstore_dataset limit 10;

Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
1,CA-2016-152156,11-08-2016,11-11-2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.96,2,0.0,41.9136
2,CA-2016-152156,11-08-2016,11-11-2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs, Rounded Back",731.94,3,0.0,219.582
3,CA-2016-138688,06-12-2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters by Universal,14.62,2,0.0,6.8714
4,US-2015-108966,10-11-2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.031
5,US-2015-108966,10-11-2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.368,2,0.2,2.5164
6,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,FUR-FU-10001487,Furniture,Furnishings,"Eldon Expressions Wood and Plastic Desk Accessories, Cherry Wood",48.86,7,0.0,14.1694
7,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,OFF-AR-10002833,Office Supplies,Art,Newell 322,7.28,4,0.0,1.9656
8,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,TEC-PH-10002275,Technology,Phones,Mitel 5320 IP Phone VoIP phone,907.152,6,0.2,90.7152
9,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,OFF-BI-10003910,Office Supplies,Binders,DXL Angle-View Binders with Locking Rings by Samsill,18.504,3,0.2,5.7825
10,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,OFF-AP-10002892,Office Supplies,Appliances,Belkin F5C206VTEL 6 Outlet Surge,114.9,5,0.0,34.47


In [0]:
select count(*) as Total_rows from superstore_dataset;

Total_rows
9994


## 2.Explore table (schema, sample data).

#### Query 1: Table Schema 


In [0]:
DESCRIBE superstore_dataset;

col_name,data_type,comment
Row ID,bigint,null
Order ID,string,null
Order Date,string,null
Ship Date,string,null
Ship Mode,string,null
Customer ID,string,null
Customer Name,string,null
Segment,string,null
Country,string,null
City,string,null


Insight:
- The dataset contains customer, product, sales, shipping, and profit-related columns.
- Key business metrics include Sales, Quantity, Discount, and Profit.

#### Query 2: Sample Records 

In [0]:
SELECT *
FROM superstore_dataset
LIMIT 10;

Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
1,CA-2016-152156,11-08-2016,11-11-2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.96,2,0.0,41.9136
2,CA-2016-152156,11-08-2016,11-11-2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,Kentucky,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs, Rounded Back",731.94,3,0.0,219.582
3,CA-2016-138688,06-12-2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,California,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters by Universal,14.62,2,0.0,6.8714
4,US-2015-108966,10-11-2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.031
5,US-2015-108966,10-11-2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,Florida,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.368,2,0.2,2.5164
6,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,FUR-FU-10001487,Furniture,Furnishings,"Eldon Expressions Wood and Plastic Desk Accessories, Cherry Wood",48.86,7,0.0,14.1694
7,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,OFF-AR-10002833,Office Supplies,Art,Newell 322,7.28,4,0.0,1.9656
8,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,TEC-PH-10002275,Technology,Phones,Mitel 5320 IP Phone VoIP phone,907.152,6,0.2,90.7152
9,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,OFF-BI-10003910,Office Supplies,Binders,DXL Angle-View Binders with Locking Rings by Samsill,18.504,3,0.2,5.7825
10,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,OFF-AP-10002892,Office Supplies,Appliances,Belkin F5C206VTEL 6 Outlet Surge,114.9,5,0.0,34.47


Insight:
- Displayed the first 10 records from the dataset.
- Helps to verify that data has been loaded correctly and understand the structure of each row.

#### Query 3: Counting Total Records

In [0]:
SELECT COUNT(*) AS Total_Rows
FROM superstore_dataset;

Total_Rows
9994


Insight:

- Shows the total number of rows available.
- There are 9994 row are available in table.``

## 3.Apply WHERE filters (region, category, date, sales). 

#### Query 1:Technology Products Sold in the West Region 

In [0]:
SELECT * FROM superstore_dataset WHERE Category='Technology' and Region = 'West' limit 10;

Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
8,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,TEC-PH-10002275,Technology,Phones,Mitel 5320 IP Phone VoIP phone,907.152,6,0.2,90.7152
12,CA-2014-115812,06-09-2014,6/14/2014,Standard Class,BH-11710,Brosina Hoffman,Consumer,United States,Los Angeles,California,90032,West,TEC-PH-10002033,Technology,Phones,Konftel 250 Conference�phone�- Charcoal black,911.424,4,0.2,68.3568
20,CA-2014-143336,8/27/2014,09-01-2014,Second Class,ZD-21925,Zuschuss Donatelli,Consumer,United States,San Francisco,California,94109,West,TEC-PH-10001949,Technology,Phones,Cisco SPA 501G IP Phone,213.48,3,0.2,16.011
27,CA-2016-121755,1/16/2016,1/20/2016,Second Class,EH-13945,Eric Hoffmann,Consumer,United States,Los Angeles,California,90049,West,TEC-AC-10003027,Technology,Accessories,Imation�8GB Mini TravelDrive USB 2.0�Flash Drive,90.57,3,0.0,11.7741
63,CA-2015-135545,11/24/2015,11/30/2015,Standard Class,KM-16720,Kunst Miller,Consumer,United States,Los Angeles,California,90004,West,TEC-AC-10004633,Technology,Accessories,"Verbatim 25 GB 6x Blu-ray Single Layer Recordable Disc, 3/Pack",13.98,2,0.0,6.1512
69,CA-2014-106376,12-05-2014,12-10-2014,Standard Class,BS-11590,Brendan Sweed,Corporate,United States,Gilbert,Arizona,85234,West,TEC-PH-10002726,Technology,Phones,netTALK DUO VoIP Telephone Service,167.968,4,0.2,62.988
91,CA-2016-109806,9/17/2016,9/22/2016,Standard Class,JS-15685,Jim Sink,Corporate,United States,Los Angeles,California,90036,West,TEC-PH-10004093,Technology,Phones,Panasonic Kx-TS550,73.584,2,0.2,8.2782
104,US-2015-156867,11/13/2015,11/17/2015,Standard Class,LC-16870,Lena Cacioppo,Consumer,United States,Aurora,Colorado,80013,West,TEC-AC-10001552,Technology,Accessories,Logitech K350 2.4Ghz Wireless Keyboard,238.896,6,0.2,-26.8758
153,CA-2016-158834,3/13/2016,3/16/2016,First Class,TW-21025,Tamara Willingham,Home Office,United States,Scottsdale,Arizona,85254,West,TEC-PH-10001254,Technology,Phones,Jabra BIZ 2300 Duo QD Duo Corded�Headset,203.184,2,0.2,15.2388
187,CA-2014-123260,8/26/2014,8/30/2014,Standard Class,FM-14290,Frank Merwin,Home Office,United States,Los Angeles,California,90032,West,TEC-AC-10002323,Technology,Accessories,SanDisk Ultra 32 GB MicroSDHC Class 10 Memory Card,176.8,8,0.0,22.984


Insight:

- Analyzes demand for technology products in a specific region.
- Helps identify regional product preferences.

#### Query 2:High-Value Orders with Good Profit

In [0]:
SELECT `Order ID`,`Customer Name`,`Product Name`,Sales,Profit
FROM superstore_dataset
WHERE Sales > 1000 AND Profit > 100
ORDER BY Sales DESC limit 10;

Order ID,Customer Name,Product Name,Sales,Profit
CA-2016-118689,Tamara Chand,Canon imageCLASS 2200 Advanced Copier,17499.95,8399.976
CA-2017-140151,Raymond Buch,Canon imageCLASS 2200 Advanced Copier,13999.96,6719.9808
CA-2017-127180,Tom Ashbrook,Canon imageCLASS 2200 Advanced Copier,11199.968,3919.9888
CA-2017-166709,Hunter Lopez,Canon imageCLASS 2200 Advanced Copier,10499.97,5039.9856
CA-2016-117121,Adrian Barton,GBC Ibimaster 500 Manual ProClick Binding System,9892.74,4946.37
CA-2014-116904,Sanjit Chand,Ibico EPK-21 Electric Binding System,9449.95,4630.4755
US-2016-107440,Bill Shonely,"3D Systems Cube Printer, 2nd Generation, Magenta",9099.93,2365.9818
CA-2016-158841,Sanjit Engle,"HP Designjet T520 Inkjet Large Format Printer - 24"" Color",8749.95,2799.984
CA-2016-143714,Christopher Conant,Canon imageCLASS 2200 Advanced Copier,8399.976,1119.9968
CA-2014-143917,Ken Lonsdale,High Speed Automatic Electric Letter Opener,8187.65,327.506


Insight:

Identifies premium orders that generated substantial revenue and profit.


#### **Query** 3:Orders with Heavy Discounts

In [0]:
SELECT `Product Name`,Sales,Discount,Profit FROM superstore_dataset
WHERE Discount >= 0.4 ORDER BY Discount DESC limit 10;

Product Name,Sales,Discount,Profit
"Acco 7-Outlet Masterpiece Power Center, Wihtout Fax/Phone Line Protection",97.264,0.8,-243.16
Kensington 7 Outlet MasterPiece Power Center,177.98,0.8,-453.849
Avery Hidden Tab Dividers for Binding Systems,1.788,0.8,-3.0396
Eureka Disposable Bags for Sanitaire Vibra Groomer I Upright Vac,1.624,0.8,-4.466
Storex DuraTech Recycled Plastic Frosted Binders,2.544,0.8,-3.816
Kensington 7 Outlet MasterPiece HOMEOFFICE Power Control Center,52.448,0.8,-131.12
"Holmes Replacement Filter for HEPA Air Cleaner, Very Large Room, HEPA Filter",68.81,0.8,-123.858
Economy Binders,1.248,0.8,-1.9344
Eureka Sanitaire Commercial Upright,66.284,0.8,-178.9668
Round Ring Binders,2.08,0.8,-3.432


Insight:

- Highlights products with more  discounts.
- Useful for evaluating the impact of discount strategies.

#### Query 4:Fast Shipping for Expensive Orders 

In [0]:
SELECT `Order ID`,`Ship Mode`,Sales,Profit
FROM superstore_dataset
WHERE `Ship Mode`='First Class'AND Sales>500 limit 10;

Order ID,Ship Mode,Sales,Profit
CA-2016-117590,First Class,1097.544,123.4737
CA-2015-102281,First Class,899.136,112.392
CA-2015-102281,First Class,626.352,46.9764
US-2016-141544,First Class,786.744,-258.5016
CA-2016-129714,First Class,4355.168,1415.4296
CA-2017-155698,First Class,647.84,168.4384
CA-2016-127369,First Class,714.3,207.147
CA-2017-140963,First Class,514.165,-30.245
CA-2014-154627,First Class,2735.952,341.994
CA-2014-112158,First Class,883.92,-110.49


Insight:
- Shows high value orders shipped using premium shipping methods.
- Helps evaluate customer service for important orders.

#### Query 5:Top Customers Generating Large Orders

In [0]:
SELECT `Customer Name`,Sales,Profit
FROM superstore_dataset
WHERE Sales > 1000 ORDER BY Sales DESC limit 10;

Customer Name,Sales,Profit
Sean Miller,22638.48,-1811.0784
Tamara Chand,17499.95,8399.976
Raymond Buch,13999.96,6719.9808
Tom Ashbrook,11199.968,3919.9888
Hunter Lopez,10499.97,5039.9856
Adrian Barton,9892.74,4946.37
Sanjit Chand,9449.95,4630.4755
Bill Shonely,9099.93,2365.9818
Sanjit Engle,8749.95,2799.984
Christopher Conant,8399.976,1119.9968


Insight:

- shows customers involved in large value purchases.
- Can be used to identify potential VIP customers.

## 4.Use GROUP BY for aggregations (sales, quantity, averages).

#### Query 1:Total Sales and Profit by Region

In [0]:
SELECT Region,ROUND(SUM(Sales),2) AS Total_Sales,ROUND(SUM(Profit),2) AS Total_Profit
FROM superstore_dataset
GROUP BY Region ORDER BY Total_Sales DESC;

Region,Total_Sales,Total_Profit
West,725457.82,108418.45
East,678781.24,91522.78
Central,501239.89,39706.36
South,391721.91,46749.43


Insight:

- Compares sales and profitability across regions.
- Helps identify the best performing market.

#### Query 2:Total Quantity Sold by Category 


In [0]:
SELECT Category,SUM(Quantity) AS Total_Quantity
FROM superstore_dataset
GROUP BY Category ORDER BY Total_Quantity DESC;

Category,Total_Quantity
Office Supplies,22906
Furniture,8028
Technology,6939


Insight:

- Shows which product categories are most in demand.
- Useful for inventory planning.

#### Query 3:Average Sales and Average Profit by Customer Segment 

In [0]:
SELECT Segment,round(AVG(Sales),2) AS Avg_Sales,round(AVG(Profit),2) AS Avg_Profit
FROM superstore_dataset
GROUP BY Segment ORDER BY Avg_Sales DESC;

Segment,Avg_Sales,Avg_Profit
Home Office,240.97,33.82
Corporate,233.82,30.46
Consumer,223.73,25.84


Insight:

- Reveals which customer segment generates larger orders and profits.
- Supports customer targeting strategies.

#### Query 4:Top 10 States by Total Profit 

In [0]:
SELECT State,ROUND(SUM(Profit),2) AS Total_Profit
FROM superstore_dataset
GROUP BY State ORDER BY Total_Profit DESC LIMIT 10;

State,Total_Profit
California,76381.39
New York,74038.55
Washington,33402.65
Michigan,24463.19
Virginia,18597.95
Indiana,18382.94
Georgia,16250.04
Kentucky,11199.7
Minnesota,10823.19
Delaware,9977.37


Insight:

- Identifies the most profitable states.
- Helps prioritize sales efforts geographically.

#### Query 5:Sales, Profit, and Quantity by Category

In [0]:
SELECT Category,ROUND(SUM(Sales),2) AS Total_Sales,SUM(Quantity) AS Total_Quantity,
ROUND(SUM(Profit),2) AS Total_Profit FROM superstore_dataset
GROUP BY Category ORDER BY Total_Sales DESC;

Category,Total_Sales,Total_Quantity,Total_Profit
Technology,836154.03,6939,145454.95
Furniture,741999.8,8028,18451.27
Office Supplies,719047.03,22906,122490.8


Insight:

- Provides a complete business view of each category.
- This  identifies which product categories contribute the most to revenue and profit.

## 5.Sort and limit results (top products, top categories). 

#### Query 1:Top 2 Categories by Sales 

In [0]:
SELECT Category,ROUND(SUM(Sales),2) AS Total_Sales
FROM superstore_dataset
GROUP BY Category ORDER BY Total_Sales DESC LIMIT 2;

Category,Total_Sales
Technology,836154.03
Furniture,741999.8


Insight:

- Reveals which categories generate the highest revenue.
- Helps focus marketing and sales efforts on high-performing categories.

#### Query 2:Top 10 Customers by Sales

In [0]:
SELECT `Customer Name`,ROUND(SUM(Sales),2) AS Total_Sales
FROM superstore_dataset
GROUP BY `Customer Name` ORDER BY Total_Sales DESC LIMIT 10;

Customer Name,Total_Sales
Sean Miller,25043.05
Tamara Chand,19052.22
Raymond Buch,15117.34
Tom Ashbrook,14595.62
Adrian Barton,14473.57
Ken Lonsdale,14175.23
Sanjit Chand,14142.33
Hunter Lopez,12873.3
Sanjit Engle,12209.44
Christopher Conant,12129.07


#### Query 3: Top 5 States by Profit 

In [0]:
SELECT State,ROUND(SUM(Profit),2) AS Total_Profit
FROM superstore_dataset
GROUP BY State ORDER BY Total_Profit DESC LIMIT 5;

State,Total_Profit
California,76381.39
New York,74038.55
Washington,33402.65
Michigan,24463.19
Virginia,18597.95


Insight:

- Shows the most profitable geographic markets.
- Supports strategic decisions regarding regional expansion and investment.

#### Query 4:Top 10 Products with High Sales and High Profit

In [0]:
SELECT `Product Name`,ROUND(SUM(Sales),2) AS Total_Sales,ROUND(SUM(Profit),2) AS Total_Profit
FROM superstore_dataset
GROUP BY `Product Name` ORDER BY Total_Profit DESC, Total_Sales DESC LIMIT 10;

Product Name,Total_Sales,Total_Profit
Canon imageCLASS 2200 Advanced Copier,61599.82,25199.93
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind,27453.38,7753.04
Hewlett Packard LaserJet 3310 Copier,18839.69,6983.88
Canon PC1060 Personal Laser Copier,11619.83,4570.93
"HP Designjet T520 Inkjet Large Format Printer - 24"" Color",18374.9,4094.98
Ativa V4110MDD Micro-Cut Shredder,7699.89,3772.95
"3D Systems Cube Printer, 2nd Generation, Magenta",14299.89,3717.97
Plantronics Savi W720 Multi-Device Wireless Headset System,9367.29,3696.28
Ibico EPK-21 Electric Binding System,15875.92,3345.28
Zebra ZM400 Thermal Label Printer,6965.7,3343.54


Insight:

Identifies products that good in both revenue generation and profitability.

#### Query 5:Top 10 Loss-Making Products by Sales 

In [0]:
SELECT `Product Name`,ROUND(SUM(Sales),2) AS Total_Sales,ROUND(SUM(Profit),2) AS Total_Profit
FROM superstore_dataset
GROUP BY `Product Name` HAVING SUM(Profit) < 0 ORDER BY Total_Sales DESC LIMIT 10;

Product Name,Total_Sales,Total_Profit
Cisco TelePresence System EX90 Videoconferencing Unit,22638.48,-1811.08
GBC DocuBind P400 Electric Binding System,17965.07,-1878.17
High Speed Automatic Electric Letter Opener,17030.31,-262.0
Lexmark MX611dhe Monochrome Laser Printer,16829.9,-4589.97
Martin Yale Chadless Opener Electric Letter Opener,16656.2,-1299.18
"Riverside Palais Royal Lawyers Bookcase, Royale Cherry Finish",15610.97,-669.54
Bretford Rectangular Conference Table Tops,12995.29,-327.23
Cubify CubeX 3D Printer Double Head Print,11099.96,-8879.97
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases,9917.64,-2876.12
Bush Advantage Collection Racetrack Conference Table,9544.73,-1934.4


Insight:
- Identifies products generating significant revenue but resulting in overall losses.
- Indicates potential issues with pricing, discount policies, or operational costs.

## 6.Solve use cases (monthly trends, top customers, duplicates)

#### Query 1:Monthly Sales Trend 

In [0]:
SELECT
    YEAR(
        COALESCE(
            try_to_timestamp(`Order Date`, 'MM-dd-yyyy'),
            try_to_timestamp(`Order Date`, 'M/d/yyyy')
        )
    ) AS Year,
    MONTH(
        COALESCE(
            try_to_timestamp(`Order Date`, 'MM-dd-yyyy'),
            try_to_timestamp(`Order Date`, 'M/d/yyyy')
        )
    ) AS Month,
    ROUND(SUM(Sales),2) AS Total_Sales
FROM superstore_dataset
GROUP BY
    YEAR(
        COALESCE(
            try_to_timestamp(`Order Date`, 'MM-dd-yyyy'),
            try_to_timestamp(`Order Date`, 'M/d/yyyy')
        )
    ),
    MONTH(
        COALESCE(
            try_to_timestamp(`Order Date`, 'MM-dd-yyyy'),
            try_to_timestamp(`Order Date`, 'M/d/yyyy')
        )
    )
ORDER BY Year, Month;

Year,Month,Total_Sales
2014,1,14236.89
2014,2,4519.89
2014,3,55691.01
2014,4,28295.34
2014,5,23648.29
2014,6,34595.13
2014,7,33946.39
2014,8,27909.47
2014,9,81777.35
2014,10,31453.39


Insight:
- Tracks sales performance over time and identifies seasonal trends.
- Helps determine peak and low-performing months for business planning.
- Useful for forecasting future demand and revenue.

#### Query 2:Top 10 Customers by Revenue 

In [0]:
SELECT `Customer Name`,COUNT(DISTINCT `Order ID`) AS Total_Orders,ROUND(SUM(Sales),2) AS Total_Sales,
ROUND(SUM(Profit),2) AS Total_Profit FROM superstore_dataset
GROUP BY `Customer Name` ORDER BY Total_Sales DESC LIMIT 10;

Customer Name,Total_Orders,Total_Sales,Total_Profit
Sean Miller,5,25043.05,-1980.74
Tamara Chand,5,19052.22,8981.32
Raymond Buch,6,15117.34,6976.1
Tom Ashbrook,4,14595.62,4703.79
Adrian Barton,10,14473.57,5444.81
Ken Lonsdale,12,14175.23,806.86
Sanjit Chand,9,14142.33,5757.41
Hunter Lopez,6,12873.3,5622.43
Sanjit Engle,11,12209.44,2650.68
Christopher Conant,5,12129.07,2177.05


Insight:
- Identifies the most valuable customers based on spending and profitability.

#### Query 3:Duplicate Orders

In [0]:
SELECT `Order ID`,COUNT(*) AS Records_Per_Order
FROM superstore_dataset
GROUP BY `Order ID` HAVING COUNT(*) > 1 ORDER BY Records_Per_Order DESC;

Order ID,Records_Per_Order
CA-2017-100111,14
CA-2017-157987,12
US-2016-108504,11
CA-2016-165330,11
US-2015-126977,10
CA-2015-131338,10
CA-2016-105732,10
CA-2015-164882,9
CA-2015-104346,9
CA-2015-132626,9


##  7.Validate results (row counts, data quality).

#### Query 1:Total Row Count

In [0]:
SELECT COUNT(*) AS Total_Rows FROM superstore_dataset;

Total_Rows
9994


Insight
- Confirms that the dataset was loaded successfully.
- Return the Total Number of Rows

#### Query 2:Checking for Missing Values

In [0]:
SELECT
    SUM(CASE WHEN `Order ID` IS NULL THEN 1 ELSE 0 END) AS Missing_OrderID,
    SUM(CASE WHEN `Customer ID` IS NULL THEN 1 ELSE 0 END) AS Missing_CustomerID,
    SUM(CASE WHEN `Product ID` IS NULL THEN 1 ELSE 0 END) AS Missing_ProductID,
    SUM(CASE WHEN Sales IS NULL THEN 1 ELSE 0 END) AS Missing_Sales,
    SUM(CASE WHEN Profit IS NULL THEN 1 ELSE 0 END) AS Missing_Profit
FROM superstore_dataset;

Missing_OrderID,Missing_CustomerID,Missing_ProductID,Missing_Sales,Missing_Profit
0,0,0,0,0


Insight
- Verifies data completeness.
- Missing values in key columns can affect analysis accuracy.

#### Query 3:Checking for Duplicate Rows

In [0]:
SELECT *, COUNT(*) AS Duplicate_Count
FROM superstore_dataset
GROUP BY ALL HAVING COUNT(*) > 1;

Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Duplicate_Count


Insight
- Detects fully duplicated records.
- Ensures sales and profit are not counted multiple times.

#### Query 4:Checking for Invalid Sales or Quantity


In [0]:
SELECT *
FROM superstore_dataset WHERE Sales <= 0 OR Quantity <= 0;

Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit


Insight
- Identifies potentially incorrect transactions.
- Negative or zero sales/quantity values may indicate data entry issues.

#### Query 5:Checking for Orders Shipped Before They Were Placed

In [0]:
SELECT * FROM superstore_dataset
WHERE `Ship Date` < `Order Date`;

Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,State,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
19,CA-2014-143336,8/27/2014,09-01-2014,Second Class,ZD-21925,Zuschuss Donatelli,Consumer,United States,San Francisco,California,94109,West,OFF-AR-10003056,Office Supplies,Art,Newell 341,8.56,2,0.0,2.4824
20,CA-2014-143336,8/27/2014,09-01-2014,Second Class,ZD-21925,Zuschuss Donatelli,Consumer,United States,San Francisco,California,94109,West,TEC-PH-10001949,Technology,Phones,Cisco SPA 501G IP Phone,213.48,3,0.2,16.011
21,CA-2014-143336,8/27/2014,09-01-2014,Second Class,ZD-21925,Zuschuss Donatelli,Consumer,United States,San Francisco,California,94109,West,OFF-BI-10002215,Office Supplies,Binders,"Wilson Jones Hanging View Binder, White, 1""",22.72,4,0.2,7.384
67,US-2015-164175,4/30/2015,05-05-2015,Standard Class,PS-18970,Paul Stevenson,Home Office,United States,Chicago,Illinois,60610,Central,FUR-CH-10001146,Furniture,Chairs,"Global Value Mid-Back Manager's Chair, Gray",213.115,5,0.3,-15.2225
73,US-2015-134026,4/26/2015,05-02-2015,Standard Class,JE-15745,Joel Eaton,Consumer,United States,Memphis,Tennessee,38109,South,FUR-CH-10000513,Furniture,Chairs,High-Back Leather Manager's Chair,831.936,8,0.2,-114.3912
74,US-2015-134026,4/26/2015,05-02-2015,Standard Class,JE-15745,Joel Eaton,Consumer,United States,Memphis,Tennessee,38109,South,FUR-FU-10003708,Furniture,Furnishings,"Tenex Traditional Chairmats for Medium Pile Carpet, Standard Lip, 36"" x 48""",97.04,2,0.2,1.213
75,US-2015-134026,4/26/2015,05-02-2015,Standard Class,JE-15745,Joel Eaton,Consumer,United States,Memphis,Tennessee,38109,South,OFF-ST-10004123,Office Supplies,Storage,Safco Industrial Wire Shelving System,72.784,1,0.2,-18.196
93,CA-2015-149587,1/31/2015,02-05-2015,Second Class,KB-16315,Karl Braun,Consumer,United States,Minneapolis,Minnesota,55407,Central,OFF-PA-10003177,Office Supplies,Paper,Xerox 1999,12.96,2,0.0,6.2208
94,CA-2015-149587,1/31/2015,02-05-2015,Second Class,KB-16315,Karl Braun,Consumer,United States,Minneapolis,Minnesota,55407,Central,FUR-FU-10003799,Furniture,Furnishings,"Seth Thomas 13 1/2"" Wall Clock",53.34,3,0.0,16.5354
95,CA-2015-149587,1/31/2015,02-05-2015,Second Class,KB-16315,Karl Braun,Consumer,United States,Minneapolis,Minnesota,55407,Central,OFF-BI-10002852,Office Supplies,Binders,Ibico Standard Transparent Covers,32.96,2,0.0,16.1504


Insight
- Verifies the logical consistency of order and shipping dates.